### DISCUSSION TITLE
waitUntil Basic Example not working?

### DISCUSSION THREAD

**Discussion by f-prettyland** [vercel/community#1947] [Help] [answered]
### Summary

tldr; waitUntil with @vercel/edge does not work in a simple example case.

I'm attempting to use waitUntil - I wanted to follow the basic example from the @vercel/edge Reference [[link](https://vercel.com/docs/concepts/functions/edge-functions/vercel-edge-package#requestcontext)].
I've made a small example repo [[link](https://github.com/f-prettyland/waitUntil-Bug)], the relevant middleware.ts can be found below as well.

Expected behaviour:
After I run vercel dev (running the server locally), I can query for the api/test page I've made.
Both the start and done message should be seen in the server side console.

Actual behaviour:
After I run vercel dev (running the server locally), I can query for the api/test page I've made.
Only the start message is seen in the server side console.

---

I'm still new to vercel, so it might easily be the case that I'm missing something obvious!

### Additional information

```
import { next } from '@vercel/edge'
import type { RequestContext } from '@vercel/edge';

export const config = {
  runtime: 'edge',
};

async function doWait() {
  const res = { foo: 'bar' }
  await wait(1000);
  console.log('!!!done!!!')
  return res
}
 
const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
 
export default function middleware(req: Request, context: RequestContext) {
  console.log('!!!start!!!')
  context.waitUntil(doWait().then((json) => console.log({ json })))

  //nop
  return next();
}
```

### Example

https://github.com/f-prettyland/waitUntil-Bug

**Comment by f-prettyland** [comment, marked as answer]
Appears this only happens in my local environment, but when deployed - no longer has this behaviour...

**Reply by f-prettyland** [reply]
To be clear, if you're having middleware issues --> try it on a remote server (`vercel deploy`) rather than locally (`vercel dev`)!
